dbms/src/Storages/StorageFile.cpp

```diff
@@ -237,10 +240,29 @@ class StorageFileBlockInputStream : public IBlockInputStream
 
     Block readImpl() override
     {
-        return reader->read();
+        auto res = reader->read();
+        if (res && file_path)
```

Comment:
We should also check if virtual columns were asked in read() query for storage. Otherwise no need to add them.